datafusion/core/tests/sql/subqueries.rs

```diff
@@ -288,6 +288,136 @@ async fn tpch_q17_correlated() -> Result<()> {
+        Inner Join: #lineitem.l_partkey = #part.p_partkey
+          Filter: #lineitem.l_shipmode IN ([Utf8("AIR"), Utf8("AIR REG")]) AND #lineitem.l_shipinstruct = Utf8("DELIVER IN PERSON")
+            TableScan: lineitem projection=[l_partkey, l_quantity, l_extendedprice, l_discount, l_shipinstruct, l_shipmode], partial_filters=[#lineitem.l_shipmode IN ([Utf8("AIR"), Utf8("AIR REG")]), #lineitem.l_shipinstruct = Utf8("DELIVER IN PERSON")]
+          TableScan: part projection=[p_partkey, p_brand, p_size, p_container]"#
+        .to_string();
+    assert_eq!(actual, expected);
+
+    Ok(())
+}
+
+// Fix for issue#78 join predicates from inside of OR expr also pulled up properly.
+#[tokio::test]
+async fn tpch_q19_pull_predicates_to_innerjoin_simplified() -> Result<()> {
+    let ctx = SessionContext::new();
+
+    register_tpch_csv(&ctx, "part").await?;
+    register_tpch_csv(&ctx, "lineitem").await?;
+    register_tpch_csv(&ctx, "partsupp").await?;
+
+    let sql = r#"
+select
+    p_partkey,
+    sum(l_extendedprice),
+    avg(l_discount),
+    count(distinct ps_suppkey)
+from
+    lineitem,
+    part,
+    partsupp
+where
+    (
+                p_partkey = l_partkey
+            and p_brand = 'Brand#12'
+            and p_partkey = ps_partkey
+        )
+   or
+    (
+                ps_partkey = p_partkey
+            and p_brand = 'Brand#23'
+            and p_partkey = l_partkey
+        )
+
+        group by p_partkey
+        ;"#;
+
+    // assert plan
+    let plan = ctx.create_logical_plan(sql).unwrap();
+    debug!("input:\n{}", plan.display_indent());
+
+    let plan = ctx.optimize(&plan).unwrap();
+    let actual = format!("{}", plan.display_indent());
+    let expected = r#"Projection: #part.p_partkey, #SUM(lineitem.l_extendedprice), #AVG(lineitem.l_discount), #COUNT(DISTINCT partsupp.ps_suppkey)
+  Aggregate: groupBy=[[#part.p_partkey]], aggr=[[SUM(#lineitem.l_extendedprice), AVG(#lineitem.l_discount), COUNT(DISTINCT #partsupp.ps_suppkey)]]
+    Inner Join: #part.p_partkey = #partsupp.ps_partkey
+      Inner Join: #lineitem.l_partkey = #part.p_partkey
+        TableScan: lineitem projection=[l_partkey, l_extendedprice, l_discount]
+        Filter: #part.p_brand = Utf8("Brand#12") OR #part.p_brand = Utf8("Brand#23")
+          TableScan: part projection=[p_partkey, p_brand], partial_filters=[#part.p_brand = Utf8("Brand#12") OR #part.p_brand = Utf8("Brand#23")]
+      TableScan: partsupp projection=[ps_partkey, ps_suppkey]"#
+        .to_string();
+    assert_eq!(actual, expected);
+
+    // assert data
+    let results = execute_to_batches(&ctx, sql).await;
+    let expected = vec![
+       "+-----------+-------------------------------+--------------------------+-------------------------------------+",
```

Comment:
We now have the ability to specify test data isolated to this particular test. It might be good to use custom test data to show a result vs just empty. See the `tpch_q4_correlated()` test for an example.

Reply:
ok, I have committed tests with tempdata as you have suggested. 
`    let expected = vec![
        "+-----------+-------------------------------+--------------------------+-------------------------------------+",
        "| p_partkey | SUM(lineitem.l_extendedprice) | AVG(lineitem.l_discount) | COUNT(DISTINCT partsupp.ps_suppkey) |",
        "+-----------+-------------------------------+--------------------------+-------------------------------------+",
        "| 63700     | 13309.6                       | 0.1                      | 1                                   |",
        "+-----------+-------------------------------+--------------------------+-------------------------------------+",
    ];
`